webapp/channels/src/components/post_emoji/post_emoji.tsx

```diff
@@ -13,23 +13,23 @@ declare module 'react' {
     }
 }
 
-export default class PostEmoji extends React.PureComponent<PostEmojiProps> {
-    public render() {
-        const emojiText = ':' + this.props.name + ':';
+const PostEmoji = ({name, imageUrl}: PostEmojiProps): JSX.Element => {
+    const emojiText = ':' + name + ':';
 
-        if (!this.props.imageUrl) {
-            return emojiText;
-        }
-
-        return (
-            <span
-                alt={emojiText}
-                className='emoticon'
-                title={emojiText}
-                style={{backgroundImage: 'url(' + this.props.imageUrl + ')'}}
-            >
-                {emojiText}
-            </span>
-        );
+    if (!imageUrl) {
+        return <span>{emojiText}</span>;
```

Comment:
is wrapping in span needed? i would be wary it this induces any styling changes

Reply:
If that span is needed for this to fit the `JSX.Element` return type, it'd be better to change this to return `<>{emojiText}</>` to avoid adding an extra span to the DOM.

I went down a rabbit hole reading about why TypeScript doesn't allow a functional component to return a string like React itself does, and it turns out there's at least 5 years of people complaining about that exact thing. While it'd be nice to just return the string, it seems like there's a long history of people asking for that, and since it hasn't happened, it's probably best to just wrap the text in a Fragment